src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -2580,6 +2580,36 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::FromOtDnsRespons
     return CHIP_NO_ERROR;
 }
 
+template <class ImplClass>
+void GenericThreadStackManagerImpl_OpenThread<ImplClass>::ResolveAddress(intptr_t context, otDnsAddressCallback callback)
+{
+    DnsResult * dnsResult = reinterpret_cast<DnsResult *>(context);
+
+    ThreadStackMgrImpl().LockThreadStack();
+
+    const otDnsQueryConfig * defaultConfig = otDnsClientGetDefaultConfig(ThreadStackMgrImpl().OTInstance());
+
+    char fullHostName[chip::Dnssd::kHostNameMaxLength + 1 + SrpClient::kDefaultDomainNameSize + 1];
+    snprintf(fullHostName, sizeof(fullHostName), "%s.%s", dnsResult->mMdnsService.mHostName, SrpClient::kDefaultDomainName);
+
+    otDnsClientResolveAddress(ThreadStackMgrImpl().OTInstance(), fullHostName, callback, reinterpret_cast<void *>(dnsResult),
```

Comment:
Should we handle the `error` here?

If `otDnsClientResolveAddress()` returns an error then the passed-in  `callback` will not be invoked (and we may need to free/clear any allocated items, like I guess the `dnsResult`?)

Reply:
Yes, thank you for catching it!. It's fixed.